virtualizarr/parsers/hdf/hdf.py

```diff
@@ -0,0 +1,349 @@
+        encoded_cf_fill_value = encode_cf_fill_value(attrs["_FillValue"], dtype)
+        attrs["_FillValue"] = encoded_cf_fill_value
+
+    codec_configs = [
+        numcodec_config_to_configurable(codec.get_config()) for codec in codecs
+    ]
+
+    fill_value = dataset.fillvalue.item()
+    dims = tuple(_dataset_dims(dataset, group=group))
+    metadata = create_v3_array_metadata(
+        shape=dataset.shape,
+        data_type=dtype,
+        chunk_shape=chunks,
+        fill_value=fill_value,
+        codecs=codec_configs,
+        dimension_names=dims,
+        attributes=attrs,
+    )
+    manifest = _dataset_chunk_manifest(filepath, dataset)
+    return ManifestArray(metadata=metadata, chunkmanifest=manifest)
+
+
+def _construct_manifest_group(
+    filepath: str,
+    reader: ObstoreReader,
+    *,
+    group: str | None = None,
+    drop_variables: Optional[Iterable[str]] = None,
+) -> ManifestGroup:
+    """
+    Construct a virtual Group from a HDF dataset.
+    """
+
+    if drop_variables is None:
+        drop_variables = []
+
+    f = h5py.File(reader, mode="r")
+
+    if group is not None and group != "":
+        g = f[group]
+        group_name = group
+        if not isinstance(g, h5py.Group):
+            raise ValueError("The provided group is not an HDF group")
+    else:
+        g = f["/"]
+        group_name = "/"
+
+    manifest_dict = {}
+    # Several of our test fixtures which use xr.tutorial data have
+    # non coord dimensions serialized using big endian dtypes which are not
+    # yet supported in zarr-python v3.  We'll drop these variables for the
+    # moment until big endian support is included upstream.)
+
+    non_coordinate_dimension_vars = _find_non_coord_dimension_vars(group=g)
+    drop_variables = list(set(list(drop_variables) + non_coordinate_dimension_vars))
+    attrs = _extract_attrs(g)
+    for key in g.keys():
+        if key not in drop_variables:
+            if isinstance(g[key], h5py.Dataset):
+                variable = _construct_manifest_array(
+                    filepath=filepath,
+                    dataset=g[key],
+                    group=group_name,
+                )
+                if variable is not None:
+                    manifest_dict[key] = variable
+    return ManifestGroup(arrays=manifest_dict, attributes=attrs)
+
+
+class Parser:
+    def __init__(
+        self,
+        group: str | None = None,
+        drop_variables: Iterable[str] | None = None,
+    ):
+        self.group = group
+        self.drop_variables = drop_variables
+
+    def __call__(
+        self,
+        file_url: str,
+        object_store: ObjectStore,
+    ) -> ManifestStore:
+        if h5py is None:
+            raise ImportError("h5py is required for using the hdf parser")
+        # Create a group containing dataset level metadata and all the manifest arrays
+
+        filename = os.path.basename(file_url)
+        reader = ObstoreReader(store=object_store, path=filename)
```

Comment:
```suggestion
        path = urlparse(file_url).path
        reader = ObstoreReader(store=object_store, path=path)
```

Reply:
@maxrjones With this change all the tests using a local file path then fail.